### PR TITLE
feat(elasticbeanstalk): Add new service ElasticBeanstalk

### PR DESCRIPTION
**Pull request by MarioRgzLpz**
### Context

In order to cover more checks, we need to create a new AWS service for Elastic Beanstalk. This new service will allow for the implementation of additional checks.

### Description

ElasticBeanstalk service with respective unit tests.

### Checklist

- Are there new checks included in this PR? No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py
Comment (MrCloudSec) [inline prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py:1]: Please, add the tags to the environment.

File: prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                elif (
+                    option["Namespace"] == "aws:elasticbeanstalk:managedactions"
+                    and option["OptionName"] == "ManagedActionsEnabled"
+                ):
+                    environment.managed_actions_option = option
+                elif (
+                    option["Namespace"] == "aws:elasticbeanstalk:cloudwatch:logs"
+                    and option["OptionName"] == "StreamLogs"
+                ):
+                    environment.cloudwatch_option = option
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py]: Why we don't store only the value of the option?

File: prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py
Comment (MarioRgzLpz) [reply prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py:1]: Added here [7f64d1f](https://github.com/prowler-cloud/prowler/pull/5322/commits/7f64d1fe6669249b70116abd73c310335c4cda4d)

File: prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py (hunk 1 above)
Comment (MarioRgzLpz) [reply prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py]: Thought about it at first. Added here [7f64d1f](https://github.com/prowler-cloud/prowler/pull/5322/commits/7f64d1fe6669249b70116abd73c310335c4cda4d)

File: prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
+from typing import Optional
+
+from pydantic import BaseModel
+
+from prowler.lib.logger import logger
+from prowler.lib.scan_filters.scan_filters import is_resource_filtered
+from prowler.providers.aws.lib.service.service import AWSService
+
+
+################## ElasticBeanstalk
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py]:
```suggestion
```

File: prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            )
+
+
+class Environment(BaseModel):
+    id: str
+    name: str
+    arn: str
+    region: str
+    application_name: str
+    health_option: Optional[str]
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py]:
```suggestion
    health_reporting: Optional[str]
```

File: prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    managed_actions_option: Optional[str]
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py]:
```suggestion
    managed_platform_updates: Optional[str]
```

File: prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    cloudwatch_option: Optional[str]
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py]:
```suggestion
    cloudwatch_stream_logs: Optional[str]
```

File: prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                    option["Namespace"] == "aws:elasticbeanstalk:cloudwatch:logs"
+                    and option["OptionName"] == "StreamLogs"
+                ):
+                    environment.cloudwatch_option = option.get("Value", "false")
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+    def _list_tags_for_resource(self):
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service.py]:
```suggestion
    def _list_tags_for_resource(self, resource: any):
```

File: tests/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service_test.py
Comment (MrCloudSec) [inline tests/providers/aws/services/elasticbeanstalk/elasticbeanstalk_service_test.py:1]: Please, add a test for `_list_tags_for_resource` function.

**Review by MrCloudSec — CHANGES_REQUESTED** [review]

**Comment by MarioRgzLpz** [comment]
@sergargar All requested changes done in [fab3060](https://github.com/prowler-cloud/prowler/pull/5322/commits/fab3060655f85faf207c4da73ebd848442a01ea6)

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**